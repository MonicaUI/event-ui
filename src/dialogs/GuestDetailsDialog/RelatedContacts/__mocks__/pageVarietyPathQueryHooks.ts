import { useQuery, QueryResult } from '@apollo/client';
import gql from 'graphql-tag';

export const gqlQuery = gql`
  query PageVarietyPath {
    event {
      registrationPath {
        guestRegistration {
          addGuestFromRelatedContacts {
            style {
              relatedContactsModalStyles
            }
            relatedContactsModalText
          }
        }
      }
    }
  }
`;

/**
 * Generates a custom hook to query registration path data for a specified page variety.
 */
export const useGuestRegistrationPageVarietyPathQuery = (): QueryResult => {
  const query = useQuery(gqlQuery);
  return query;
};
