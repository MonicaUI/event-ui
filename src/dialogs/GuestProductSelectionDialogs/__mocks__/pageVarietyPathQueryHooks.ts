import { useQuery, QueryResult } from '@apollo/client';
import gql from 'graphql-tag';

export const gqlQuery = gql`
  query PageVarietyPath {
    event {
      registrationPath {
        registration {
          admissionItems {
            display {
              fees
            }
          }
          sessions {
            display {
              fees
            }
          }
        }
      }
    }
  }
`;

/**
 * Generates a custom hook to query registration path data for a specified page variety.
 */
export const useRegistrationPageVarietyPathQuery = (): QueryResult => {
  const query = useQuery(gqlQuery);
  return query;
};
