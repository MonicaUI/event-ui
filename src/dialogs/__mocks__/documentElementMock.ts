export const getMockedMessageContainer = (): $TSFixMe => {
  // Mock a DOM to play around
  document.body.innerHTML = `
    <div>
    <div name="messageContainer" tabIndex=-1>message container</div>
    </div>
  `;
  const mockMessageContainerFocus = jest.fn();
  const mockMessageContainer = document.getElementsByName('messageContainer')[0];
  mockMessageContainer.focus = mockMessageContainerFocus;
  return mockMessageContainer;
};

export const getMockedFieldInputs = (fieldNames: $TSFixMe): $TSFixMe => {
  // Mock a DOM to play around
  document.body.innerHTML = `
    <div>
    ${fieldNames.map(fieldName => `<input name=${fieldName}>input-${fieldName}</input>`)}
    </div>
  `;
  const mockInputFocus = jest.fn();
  const mockInputContainers = [];
  fieldNames.forEach(fieldName => {
    const mockInputContainer = document.getElementsByName(fieldName)[0];
    mockInputContainer.focus = mockInputFocus;
    mockInputContainers.push(mockInputContainer);
  });
  return mockInputContainers;
};
