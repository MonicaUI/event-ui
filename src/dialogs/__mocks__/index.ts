/* eslint-env jest */
import noopAction from '../../testUtils/noopAction';

export const openAlreadyRegisteredDialog = jest.fn(noopAction);

export const openDeclineRegistrationDialog = jest.fn(noopAction);

export const openStartNewRegistrationDialogFromPageLanding = jest.fn(noopAction);

export const openStartNewRegistrationDialogDuringRegistration = jest.fn(noopAction);

export const startNewRegistrationAndNavigateToRegistration = jest.fn(noopAction);

export const openEventWaitlistDialog = jest.fn(noopAction);

export const openEventTemporaryClosedErrorDialog = jest.fn(noopAction);

export const openNoAdmissionItemAvailableForRegistrationTypeDialog = jest.fn(noopAction);

export const openEventStatusDialog = jest.fn(noopAction);

export const openPrivateEventErrorDialog = jest.fn(noopAction);

export const openRegistrationInOtherTabDialog = jest.fn(noopAction);

export const openKnownErrorDialog = jest.fn(noopAction);

export const openPaymentProcessingErrorDialog = jest.fn(noopAction);

export const openInformationAlreadyUsedDialog = jest.fn(noopAction);

export const openPartialRegistrationConfirmationDialog = jest.fn(noopAction);

export const openPreviewModeWarningDialog = jest.fn(noopAction);

export const openCapacityReachedDialog = jest.fn(noopAction);

export const openGroupCancelRegistrationDialog = jest.fn(noopAction);

export const openSessionOverlapWarningDialog = jest.fn(noopAction);

export const openGroupMemberRemoveDialog = jest.fn(noopAction);

export const openSharePromptDialog = jest.fn(noopAction);

export const openConfirmationDialog = jest.fn(noopAction);

export const openGuestProductSelectionDialog = jest.fn(noopAction);

export const openGroupRegistrationTypeDialog = jest.fn(noopAction);

export const openGuestDetailsDialog = jest.fn(noopAction);

export const openGuestRemoveDialog = jest.fn(noopAction);

export const openContactPlannerDialog = jest.fn(noopAction);

export const openCancelRegistrationSuccessConfirmationDialog = jest.fn(noopAction);

export const openCancelRegistrationDialog = jest.fn(noopAction);

export const openShoulderDateApprovalWarningDialog = jest.fn(noopAction);

export const openTransactionInProcessingErrorDialog = jest.fn(noopAction);
