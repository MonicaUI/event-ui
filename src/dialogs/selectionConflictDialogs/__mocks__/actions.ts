export const returnToProcessStart = jest.fn(() => () => {});
