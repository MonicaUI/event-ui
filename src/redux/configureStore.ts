import analyticsConfig from './analytics';
import reducer from './reducer';

// using require instead of import for conditional compilation;
let createStoreWithMiddleware;
if (process.env.NODE_ENV === 'production') {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  createStoreWithMiddleware = require('nucleus-guestside-site/src/redux/prodCreateStoreWithMiddleware').default;
} else {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  createStoreWithMiddleware = require('nucleus-guestside-site/src/redux/devCreateStoreWithMiddleware').default;
}

export default function configureStore(
  initialState: $TSFixMe,
  baseAnalyticsConfig = {},
  thunkExtraArgument?: $TSFixMe
): $TSFixMe {
  return createStoreWithMiddleware(reducer, initialState, {
    analyticsConfig: analyticsConfig(initialState, baseAnalyticsConfig),
    thunkExtraArgument
  });
}
