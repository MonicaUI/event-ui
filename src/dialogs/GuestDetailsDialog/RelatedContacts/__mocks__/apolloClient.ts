import { MockedResponse } from '@apollo/client/testing';
import { gqlQuery } from './pageVarietyPathQueryHooks';

type RelatedContactsDialogMockData = {
  relatedContactsModalStyles: unknown;
  relatedContactsModalText: unknown;
};

export const getApolloClientMocks = (
  state: RelatedContactsDialogMockData
): MockedResponse<Record<string, $TSFixMe>>[] => {
  const { relatedContactsModalStyles = {}, relatedContactsModalText = {} } = state;
  return [
    {
      request: {
        query: gqlQuery
      },
      result: {
        data: {
          event: {
            registrationPath: {
              guestRegistration: {
                addGuestFromRelatedContacts: {
                  style: {
                    relatedContactsModalStyles
                  },
                  relatedContactsModalText
                }
              }
            }
          }
        }
      }
    }
  ];
};
