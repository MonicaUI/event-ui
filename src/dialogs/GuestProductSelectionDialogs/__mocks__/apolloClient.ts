import { MockedResponse } from '@apollo/client/testing';
import { gqlQuery } from './pageVarietyPathQueryHooks';

type GuestProductSelectionDialogMockData = {
  displayAdmissionItemsFees: unknown;
  displaySessionsFees: unknown;
};

export const getApolloClientMocks = (
  state: GuestProductSelectionDialogMockData
): MockedResponse<Record<string, $TSFixMe>>[] => {
  const { displayAdmissionItemsFees = {}, displaySessionsFees = {} } = state;
  return [
    {
      request: {
        query: gqlQuery
      },
      result: {
        data: {
          event: {
            registrationPath: {
              registration: {
                admissionItems: {
                  display: {
                    fees: displayAdmissionItemsFees
                  }
                },
                sessions: {
                  display: {
                    fees: displaySessionsFees
                  }
                }
              }
            }
          }
        }
      }
    }
  ];
};
