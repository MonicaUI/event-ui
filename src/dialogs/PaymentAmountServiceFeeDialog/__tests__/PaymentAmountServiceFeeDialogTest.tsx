import React from 'react';
import DialogContainer from 'nucleus-guestside-site/src/containers/DialogContainer';
import configureStore from '../../../redux/configureStore';
import { act } from 'react-dom/test-utils';
import { MockedProvider } from '@apollo/client/testing';
import { merge } from 'lodash';
import { openPaymentAmountServiceFeeDialog } from '..';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
// eslint-disable-next-line jest/no-mocks-import
import { RegCartClient } from '../../../widgets/PaymentWidget/__mocks__/regCartClient';
import EventSnapshot from '../../../../fixtures/EventSnapshot.json';
// eslint-disable-next-line jest/no-mocks-import
import { getApolloClientMocks } from '../../../widgets/PaymentWidget/__mocks__/apolloClient';
jest.mock('../../../widgets/PaymentWidget/getRegCartPricingAction', () => ({
  __esModule: true,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  default: async (state, _) => {
    return state;
  }
}));
jest.mock('../../../widgets/PaymentWidget/useCachedRegCartPricing', () => ({
  ...jest.requireActual<$TSFixMe>('../../../widgets/PaymentWidget/useCachedRegCartPricing'),
  __esModule: true,
  getTravelCartForQuery: () => {
    return JSON.stringify({});
  },
  lastSavedRegCartForQuery: () => {
    return JSON.stringify({});
  },
  getPaymentInfoForQuery: () => {
    return JSON.stringify({});
  }
}));

const groupMemberEventRegId = '00000000-0000-0000-0000-000000000002';
const groupLeaderEventRegId = '00000000-0000-0000-0000-000000000000';

const regCart = {
  regCartId: '928f02df-4ba7-4637-824d-ae71bd414e4f',
  status: 'INPROGRESS',
  eventRegistrations: {
    '00000000-0000-0000-0000-000000000001': {
      attendee: {
        personalInformation: {
          emailAddress: 'lroling-384934@j.mail'
        },
        attendeeId: 'aac17ae9-74b8-4687-ae9e-e2781d247f95'
      },
      confirmationNumber: '123456789',
      productRegistrations: [],
      attendeeType: 'GROUP_LEADER',
      registrationPathId: '411c6566-1e5a-4c38-b8e5-f63ab9239b40'
    },
    [groupMemberEventRegId]: {
      eventRegistrationId: '02',
      attendee: {
        personalInformation: {},
        eventAnswers: {}
      },
      attendeeType: 'ATTENDEE',
      primaryRegistrationId: groupLeaderEventRegId,
      productRegistrations: [
        {
          productId: Object.keys(EventSnapshot.eventSnapshot.products.admissionItems)[0],
          productType: 'AdmissionItem',
          quantity: 1,
          requestedAction: 'REGISTER'
        }
      ]
    }
  }
};

const initialState = {
  text: {
    translate: (resx, tokens) => (tokens ? `${resx}:${JSON.stringify(tokens)}` : resx),
    resolver: {
      date: () => 'some date',
      currency: x => x
    }
  },
  website: EventSnapshot.eventSnapshot.siteEditor.website,
  event: {
    selectedPaymentTypesSnapshot: {
      paymentMethodTypes: ['Visa', 'MasterCard']
    },
    products: {
      serviceFees: {
        'b272f019-8a00-487a-a640-938a836e74e7': {
          active: true,
          refundable: false,
          amount: 10.0,
          applyType: 0,
          adjustmentType: 1,
          inviteeType: 0,
          serviceFeeType: 51,
          applicableContactTypes: [],
          applicablePaymentMethods: ['Visa'],
          displayOrder: 2,
          code: 'how they pay - amount',
          id: 'b272f019-8a00-487a-a640-938a836e74e7',
          name: 'how they pay - amount',
          type: 'PaymentTypeServiceFee',
          defaultFeeId: 'e0e1f3ff-6c37-44cf-a454-6aca7712cea3',
          fees: {
            'e0e1f3ff-6c37-44cf-a454-6aca7712cea3': {
              chargePolicies: [
                {
                  id: 'e03a820d-8396-44d9-a95a-5e471c5ec6ed',
                  isActive: true,
                  effectiveUntil: '2999-12-31T00:00:00.000Z',
                  amount: 10.0,
                  maximumRefundAmount: 0.0
                }
              ],
              refundPolicies: [],
              isActive: true,
              isRefundable: false,
              registrationTypes: [],
              name: 'how they pay - amount',
              id: 'e0e1f3ff-6c37-44cf-a454-6aca7712cea3',
              amount: 10.0,
              glCodes: []
            }
          }
        }
      }
    },
    eventFeatureSetup: {
      fees: {
        merchantAccountId: 'merchantAccountId',
        fees: true,
        taxes: true,
        serviceFees: true
      }
    }
  },
  registrationForm: {
    regCart,
    regCartPayment: {}
  },
  clients: {
    regCartClient: new RegCartClient()
  },
  regCartStatus: {
    lastSavedRegCart: {}
  }
};

const waitWithAct = async () => {
  // Act is used because the Redux store is updated during wait
  return act(async () => {
    await new Promise(resolve => setTimeout(resolve, 0));
  });
};

let store;
const mountComponent = async (props = {}, optionalState = {}) => {
  const state = merge({}, initialState, optionalState);
  store = configureStore(state, {});

  const apolloClientMocks = getApolloClientMocks(state);

  const component = mount(
    <Provider store={store}>
      {/* @ts-expect-error ts-migrate(2322) FIXME: Type '{ children: Element; mocks: any; addTypeName... Remove this comment to see the full error message */}
      <MockedProvider mocks={apolloClientMocks} addTypeName={false}>
        <DialogContainer {...props} spinnerMessage="spinnerMessage" message="message" />
      </MockedProvider>
    </Provider>
  );
  // Wait for Apollo Client MockedProvider to render mock query results
  await waitWithAct();
  await component.update();
  return component;
};

describe('PaymentAmountServiceFeeDialog', () => {
  test('matches snapshot when opened', async () => {
    const dialog = await mountComponent({}, {});
    store.dispatch(openPaymentAmountServiceFeeDialog());
    await waitWithAct();
    await dialog.update();
    expect(dialog).toMatchSnapshot();
  });
});
