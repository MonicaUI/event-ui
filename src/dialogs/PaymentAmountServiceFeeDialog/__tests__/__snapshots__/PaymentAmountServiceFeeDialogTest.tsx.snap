// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentAmountServiceFeeDialog matches snapshot when opened 1`] = `
<div
  class=""
>
  <div
    class=""
  />
  <span
    class="mockNucleusCoreTransition"
  >
    <div
      class=""
    >
      <div
        aria-label="Dialog"
        role="dialog"
        style="outline: none;"
        tabindex="0"
      >
        <div
          class=""
          style="background-color: rgb(243, 59, 21); border-radius: 100px; position: absolute; left: 512px; top: 384px;"
        >
          <div
            class=""
          />
          <div
            class=""
          >
            <div
              class="dialogContainer"
            >
              <div
                class="dialogChildren"
              >
                <div
                  data-cvent-id="payment-amount-service-fee-dialog"
                >
                  <div
                    aria-describedby="dialogHeader"
                    class=""
                    style="background-color: rgb(221, 226, 230); border-radius: 0;"
                  >
                    <div
                      class=""
                      id="dialogHeader"
                      style="background-color: rgb(249, 32, 242); padding: 15px 1.5625rem 15px 1.5625rem;"
                    >
                      <h2
                        class=""
                        style="color: rgb(255, 255, 255); font-family: Comic Sans MS, Comic Sans, Chalkboard, Marker Felt, sans-serif; font-size: 3.5625rem; font-style: normal; font-weight: 700; line-height: 1.3;"
                      >
                        <div
                          class=""
                          data-cvent-id="close"
                          role="button"
                          tabindex="0"
                        >
                          <span
                            class="container"
                          >
                            <span
                              aria-hidden="true"
                              class=""
                              role="img"
                            />
                          </span>
                        </div>
                        ServiceFees_PaymentHeader__resx
                      </h2>
                    </div>
                  </div>
                  <div
                    class=""
                    name="messageContainer"
                    style="background-color: rgb(243, 59, 21); padding: 0px 1.25rem 1.875rem 1.25rem;"
                    tabindex="-1"
                  >
                    <div
                      class=""
                    >
                      <div
                        class="leftLineItemStyle"
                      >
                        <span
                          class="instructionalText"
                          style="font-family: PT Sans, sans-serif; font-weight: 400; font-size: 16px; color: rgb(39, 63, 105); line-height: 1.3; font-style: normal; background-color: transparent; border-radius: 0; padding: 0px 0px 0px 5px;"
                        >
                          ServiceFees_PaymentAmountType_Instruction__resx
                        </span>
                      </div>
                      <br />
                      <br />
                      <div
                        class=""
                      >
                        <br />
                        <div
                          class="oneLineItemStyle"
                          style="font-family: PT Sans, sans-serif; font-weight: 400; font-size: 16px; color: rgb(39, 63, 105); line-height: 1.3; font-style: normal; background-color: transparent; border-radius: 0; padding: 0px 0px 0px 5px;"
                        >
                          <span
                            class="additionalItemsText"
                          >
                            EventWidgets_PaymentWidget_PaymentMethodTypeVisa__resx
                          </span>
                          <span
                            class="totalAmount"
                          >
                            10
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </span>
</div>
`;
